### PR TITLE
OP-1232 Improve api security config

### PR DESCRIPTION
**Pull request by mwithi**
See OP-1232.

- [x] Replaced all deprecated methods
- [x] Added `cors.allowed.origins` parameter to better configure CORS

Cc: @massimilianoregis

### CODE REVIEW

**Review by dbmalkovsky — COMMENTED** [review]
Is this the correct indentation?  Seems like the recent ide changes moved it from 16 to 4 spaces

**Comment by mwithi** [comment]
> Is this the correct indentation? Seems like the recent ide changes moved it from 16 to 4 spaces

It is! But it seems Eclipse lacks a setting for methods indentation...

image

File: rsc/application.properties.dist [hunk 1]
```diff
@@ -13,6 +13,9 @@ server.servlet.session.cookie.http-only=true
 spring.pid.fail-on-write-error=true
 spring.pid.file=OH_API_PID
 
+### In production change to http://<domain>
+cors.allowed.origins=http://localhost:8080,http://localhost:3000
```
Comment (SilverD3) [inline rsc/application.properties.dist]: Is it possible to use env vars like in `api.host=API_HOST:API_PORT`?

File: src/main/java/org/isf/config/SecurityConfig.java [hunk 2]
```diff
@@ -88,262 +87,253 @@ public PasswordEncoder encoder() {
...
-		http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
-			.and().cors()
-			.and().csrf().disable().authorizeRequests()
-			// .expressionHandler(webExpressionHandler())
-			.and().exceptionHandling()
-			// .accessDeniedHandler(accessDeniedHandler)
-			.authenticationEntryPoint(restAuthenticationEntryPoint)
-			.and().authorizeRequests().requestMatchers("/auth/**").permitAll()
+		http
+						.cors(cors -> cors.configurationSource(corsConfigurationSource()))
```
Comment (SilverD3) [inline src/main/java/org/isf/config/SecurityConfig.java]: Indentation doesn't look correct. This is more than 4 spaces

File: rsc/application.properties.dist (hunk 1 above)
Comment (mwithi) [reply rsc/application.properties.dist]: good catch, thanks!

File: src/main/java/org/isf/config/SecurityConfig.java (hunk 2 above)
Comment (mwithi) [reply src/main/java/org/isf/config/SecurityConfig.java]: see https://github.com/informatici/openhospital-api/pull/478#issuecomment-2404734077
I'm trying to fix...

**Comment by mwithi** [comment]
> It is! But it seems Eclipse lacks a setting for methods indentation...

Found! (to be set to 1...)

image

File: src/main/java/org/isf/config/SecurityConfig.java (hunk 2 above)
Comment (mwithi) [reply src/main/java/org/isf/config/SecurityConfig.java]: Done! see https://github.com/informatici/openhospital-api/pull/478#issuecomment-2406141183

**Review by SilverD3 — CHANGES_REQUESTED** [review]
`GET /hospitals` is returning 403
oh-401

**Comment by mwithi** [comment]
> `GET /hospitals` is returning 403

Nice catch! Fixed, thanks!

**Comment by mwithi** [comment]
> Is this the correct indentation? Seems like the recent ide changes moved it from 16 to 4 spaces

fixed, thanks

**Review by SilverD3 — APPROVED** [review]
Looks good!

**Review by dbmalkovsky — APPROVED** [review]
LGTM

**Pull request merged**